test/parallel/test-overwrite-config.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+const common = require('../common');
+
+// This test ensures that overwriting a process configuration
+// value does not affect code in bootstrap_node.js.
+
+common.skipIfInspectorDisabled();
+
+const assert = require('assert');
+assert.strictEqual(process.config.variables.v8_enable_inspector, 1);
+process.config.variables.v8_enable_inspector = 100;
+assert.strictEqual(process.config.variables.v8_enable_inspector, 100);
```

Comment:
I am not sure if this is a appropriate test.
I would have expected the `v8_enable_inspector` variable to be set to something falsy because we test it to be truthy and afterwards to check if the inspector is set up nevertheless.
Another approach would be to test that inspector is set without any variable manipulation. That should already verify that manipulating it has no impact on this anymore.

Reply:
@BridgeAR Thanks, I've updated the test. When going through the code I looks like any user land code will be call by Module.runMain, at which point the checking of process.config.variables.v8_enable_inspector done by bootstrap_node.js has already been performed. This also holds true for any preloaded modules. So as far as I can tell the check should be safe unless there are other ways this could happen that I've not taken into account.